src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction-analysis.component.spec.ts

```diff
@@ -107,72 +101,60 @@ describe('ProgrammingExerciseInstructionInstructorAnalysis', () => {
-                    comp = fixture.componentInstance;
-                    debugElement = fixture.debugElement;
-
-                    exerciseHintService = debugElement.injector.get(ExerciseHintService);
-                    getHintsForExerciseStub = stub(exerciseHintService, 'findByExerciseId').returns(of({ body: exerciseHints }) as Observable<HttpResponse<ExerciseHint[]>>);
-                });
-        });
-
-        afterEach(() => {
-            getHintsForExerciseStub.restore();
-        });
-
-        it('should not render if no test cases were provided', () => {
-            comp.problemStatement = problemStatement;
-            comp.taskRegex = taskRegex;
-            triggerChanges(comp, { property: 'problemStatement', currentValue: problemStatement });
-            fixture.detectChanges();
 
-            expect(debugElement.nativeElement.innerHtml).to.be.undefined;
-            expect(comp.missingTestCases).to.be.empty;
-            expect(comp.invalidTestCases).to.be.empty;
+        describe('Analysis service integration test', () => {
+            const missingTestCases = ['test6', 'test7'];
+            const invalidTestCases = ['test3', 'test4'];
+
+            it('should not render if no test cases were provided', () => {
+                comp.problemStatement = problemStatement;
+                comp.taskRegex = taskRegex;
+                triggerChanges(comp, { property: 'problemStatement', currentValue: problemStatement });
+                fixture.detectChanges();
+
+                expect(debugElement.nativeElement.innerHtml).toBe(undefined);
+                expect(comp.missingTestCases).toEqual([]);
+                expect(comp.invalidTestCases).toEqual([]);
+            });
+
+            it('should render warnings on missing and invalid test cases', fakeAsync(() => {
+                comp.problemStatement = problemStatement;
+                comp.taskRegex = taskRegex;
+                comp.exerciseTestCases = exerciseTestCases;
+                comp.exerciseHints = exerciseHints;
+
+                const completeAnalysis = {
+                    '0': { invalidTestCases: ['artemisApp.programmingExercise.testCaseAnalysis.invalidTestCase'] },
+                    '2': {
+                        invalidTestCases: ['artemisApp.programmingExercise.testCaseAnalysis.invalidTestCase'],
+                    },
+                };
+
+                analyzeProblemStatementStub.mockReturnValue({ completeAnalysis, invalidTestCases, missingTestCases });
+
+                comp.ngOnInit();
+
+                triggerChanges(
+                    comp,
+                    { property: 'problemStatement', currentValue: problemStatement, firstChange: false },
+                    { property: 'exerciseTestCases', currentValue: exerciseTestCases },
+                );
+
+                fixture.detectChanges();
+                tick(500);
+
+                expect(debugElement.nativeElement.innerHtml).not.toEqual('');
```

Comment:
```suggestion
                expect(debugElement.nativeElement.innerHtml).not.toBe('');
```

Reply:
Can we assert something more? Some text that should be contained? Some specific element that should be part of it?

Reply:
I changed the jest command in afb6279. I also inspected the possibility to assert more specific text, but the desired state of the innerHtml is undefined (at least it was the case before these changes). So I changed the assertion to this.